crypto/asn1/tasn_utl.c

```diff
@@ -68,8 +68,8 @@ int ossl_asn1_set_choice_selector(ASN1_VALUE **pval, int value,
 int ossl_asn1_do_lock(ASN1_VALUE **pval, int op, const ASN1_ITEM *it)
 {
     const ASN1_AUX *aux;
-    CRYPTO_REF_COUNT *lck;
     CRYPTO_RWLOCK **lock;
```

Comment:
lock, including creating and freeing it can be removed.

Reply:
including the ref_lock in the ASN1_AUX structure.

Reply:
It cannot.  It's used for other purposes.
This PR avoids using it for the reference count only.

There are other similar instances.